streaming-tool-use/streaming-tool-use.ipynb

```diff
@@ -0,0 +1,551 @@
+      "Requirement already satisfied: typing-extensions<5,>=4.7 in /Users/rick/workspace/groq-api-cookbook/.venv/lib/python3.10/site-packages (from groq->-r requirements.txt (line 1)) (4.11.0)\n",
+      "Requirement already satisfied: idna>=2.8 in /Users/rick/workspace/groq-api-cookbook/.venv/lib/python3.10/site-packages (from anyio<5,>=3.5.0->groq->-r requirements.txt (line 1)) (3.7)\n",
+      "Requirement already satisfied: exceptiongroup>=1.0.2 in /Users/rick/workspace/groq-api-cookbook/.venv/lib/python3.10/site-packages (from anyio<5,>=3.5.0->groq->-r requirements.txt (line 1)) (1.2.1)\n",
+      "Requirement already satisfied: certifi in /Users/rick/workspace/groq-api-cookbook/.venv/lib/python3.10/site-packages (from httpx<1,>=0.23.0->groq->-r requirements.txt (line 1)) (2024.2.2)\n",
+      "Requirement already satisfied: httpcore==1.* in /Users/rick/workspace/groq-api-cookbook/.venv/lib/python3.10/site-packages (from httpx<1,>=0.23.0->groq->-r requirements.txt (line 1)) (1.0.5)\n",
+      "Requirement already satisfied: h11<0.15,>=0.13 in /Users/rick/workspace/groq-api-cookbook/.venv/lib/python3.10/site-packages (from httpcore==1.*->httpx<1,>=0.23.0->groq->-r requirements.txt (line 1)) (0.14.0)\n",
+      "Requirement already satisfied: annotated-types>=0.4.0 in /Users/rick/workspace/groq-api-cookbook/.venv/lib/python3.10/site-packages (from pydantic<3,>=1.9.0->groq->-r requirements.txt (line 1)) (0.6.0)\n",
+      "Requirement already satisfied: pydantic-core==2.18.2 in /Users/rick/workspace/groq-api-cookbook/.venv/lib/python3.10/site-packages (from pydantic<3,>=1.9.0->groq->-r requirements.txt (line 1)) (2.18.2)\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "%pip install -r requirements.txt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "e21816b3",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'Groq API key configured: gsk_7FdrzM...'"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import json\n",
+    "\n",
+    "from groq import Groq\n",
+    "from dotenv import load_dotenv\n",
+    "\n",
+    "load_dotenv()\n",
+    "\"Groq API key configured: \" + os.environ[\"GROQ_API_KEY\"][:10] + \"...\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7f7c9c55-e925-4cc1-89f2-58237acf14a4",
+   "metadata": {},
+   "source": [
+    "We will use the ```llama3-70b-8192``` model in this demo. Note that you will need a Groq API Key to proceed and can create an account [here](https://console.groq.com/) to generate one for free. Only Llama 3 models support parallel tool use at this time (05/07/2024).\n",
```

Comment:
I would suggest putting this intro above the imports for a better introduction/flow!